drivers/api/no_os_i2c.c

```diff
@@ -105,8 +107,12 @@ int32_t no_os_i2c_write(struct no_os_i2c_desc *desc,
 	if (!desc->platform_ops->i2c_ops_write)
 		return -ENOSYS;
 
-	return desc->platform_ops->i2c_ops_write(desc, data, bytes_number,
+	no_os_mutex_lock(desc->mutex);
+	int32_t ret = desc->platform_ops->i2c_ops_write(desc, data, bytes_number,
```

Comment:
please declare `ret` at the start of the function. There are some other places where this applies.